### DISCUSSION TITLE
Telekinetic Room QOL

### DISCUSSION THREAD

**Discussion by Jamsterman25** [runelite/runelite#18078] [Ideas]
Since the updates to the Telekinetic Grab spell, there is no longer a movement delay while the spell is traveling.
In the current edition of the MTA plugin, there is an optimal movement path that only activates when the guardian is moving and is triggered by the npc movement. 
We could instead trigger on the casting of the Telekinetic grab spell when the user is on an optimal tile and allow the user to instantly see where to move next while the spell is in the air and they have the ability to move.

Sometimes when performing the first Telekinetic grab of a room, the new optimal destination tile is shown immediately. On a quick skim of the code, I was not able to identify why this behavior was happening. The only trigger I can identify for moving and coloring the optimal tile is line 321, MAZE_GUARDIAN_MOVING. I could not find a way to identify if a spell had been cast. otherwise I would have attempted the change myself.

To implement this, we could replace the conditional at 321 MAZE_GUARDIAN_MOVING instead with: user has cast telekinetic grasp while true tile is equal to optimal position. then advance the worldpoint to the next optimal tile. This should be ok logically since telekinetic grab cant be cast while the guardian is in motion.

**Comment by geheur** [comment]
Ideally it would just show the next 2 steps at all times.

**Reply by Jamsterman25** [reply]
Yeah, just showing the next step or 2 in a different color would be great!

**Comment by Felanbird** [comment]
this was added in #18079
